models/action.js

```diff
@@ -0,0 +1,59 @@
+import mongoose from 'mongoose';
+const Schema = mongoose.Schema;
+
+const ActionSchema = new mongoose.Schema({
+  type: {
+    type: String,
+    required: true,
+    enum: ['view', 'share', 'save']
+  },
+  user: {
+    type: Schema.ObjectId,
+    ref: 'User',
+    required: true
+  },
+  entity: {
+    type: Schema.ObjectId,
+    required: true
+  },
+  entityType: {
+    type: String,
+    required: true,
+    enum: ['Post', 'Source']
+  },
+  createdAt: {
+    type: Date,
+    default: Date.now
+  },
```

Comment:
change to `created` object that contains `at` with `Date` type & `by` with `ObjectId` type ref to `User`